src/mlpack/methods/ann/init_rules/he_init.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file he_init.hpp
+ * @author Dakshit Agrawal
+ *
+ * Intialization rule given by He et. al. for neural networks. The He
+ * initialization initializes weights of the neural network to better
+ * suit the rectified activation units.
+ *
+ * For more information, the following paper can be referred to:
+ *
+ * @code
+ * @article{He2015DelvingDI,
+ * title={Delving Deep into Rectifiers: Surpassing Human-Level Performance on ImageNet Classification},
+ * author={Kaiming He and Xiangyu Zhang and Shaoqing Ren and Jian Sun},
+ * journal={2015 IEEE International Conference on Computer Vision (ICCV)},
+ * year={2015},
+ * pages={1026-1034}}
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_HE_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_HE_INIT_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+using namespace mlpack::math;
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * This class is used to initialize weight matrix with the He initialization rule.
+ */
+class HeInitialization
+{
+ public:
+    /**
+     * Initialize the HeInitialization object.
+     *
+     */
+    HeInitialization()
+    {
+        // Nothing to do here.
+    }
+
+    /**
+     * Initialize the elements of the weight matrix with the He initialization
+     * rule.
+     *
+     * @param W Weight matrix to initialize.
+     * @param rows Number of rows.
+     * @param cols Number of columns.
+     */
+    void Initialize(arma::mat& W,
+                    const size_t rows,
+                    const size_t cols)
+    {
+        double_t variance = 2 / rows;
```

Comment:
Are you sure this is correct? If I remember right, this should be ``sqrt(2/inputsize)``.

Reply:
The standard deviation is  `sqrt(2/inputsize)`.

Reply:
So, wouldn't it be more intuitive to use ``arma::randn`` so that we can use the formula from the paper?

Reply:
`arma::randn` samples from a Gaussian distribution of mean 0 and variance 1.

We want samples from a Gaussian distribution of mean 0 and variance `2/rows`.

Reply:
Right, I was thinking about the xavier initalization which uses an uniform initialization. Okay, in this case, let's go with the current form, but do you mind to add a simple comment?

Reply:
Sure, I will do that.